runelite-client/src/main/java/net/runelite/client/plugins/statusbars/BarRenderer.java

```diff
@@ -0,0 +1,170 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package net.runelite.client.plugins.statusbars;
+
+import lombok.RequiredArgsConstructor;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.BACKGROUND;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.COUNTER_ICON_HEIGHT;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.HEAL_OFFSET;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.ICON_AND_COUNTER_OFFSET_X;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.ICON_AND_COUNTER_OFFSET_Y;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.OFFSET;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.OVERHEAL_COLOR;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.OVERHEAL_OFFSET;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.PADDING;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.SKILL_ICON_HEIGHT;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.WIDTH;
+import net.runelite.client.ui.FontManager;
+import net.runelite.client.ui.overlay.components.TextComponent;
+
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.Image;
+import java.util.function.Supplier;
+
+@RequiredArgsConstructor
+class BarRenderer
+{
+	private static final TextComponent textComponent = new TextComponent();
+
+	private final StatusBarsConfig config;
+	private final Supplier<Integer> maxValueSupplier;
+	private final Supplier<Integer> currentValueSupplier;
+	private final Supplier<Integer> healSupplier;
+	private final Supplier<Color> colorSupplier;
+	private final Supplier<Color> healColorSupplier;
+	private final Supplier<Image> iconSupplier;
+
+	private int maxValue;
+	private int currentValue;
+
+	private void refreshSkills()
+	{
+		maxValue = maxValueSupplier.get();
+		currentValue = currentValueSupplier.get();
+	}
+
+	private void renderBar(Graphics2D graphics, int x, int y, int height)
+	{
+		final Color fill = colorSupplier.get();
+
+		graphics.setColor(BACKGROUND);
+		graphics.drawRect(x, y, WIDTH - PADDING, height - PADDING);
+		graphics.fillRect(x, y, WIDTH, height);
+
+		final int filledHeight = getBarHeight(maxValue, currentValue, height);
+		graphics.setColor(fill);
+		graphics.fillRect(x + PADDING,
+				y + PADDING + (height - filledHeight),
+				WIDTH - PADDING * OFFSET,
+				filledHeight - PADDING * OFFSET);
+	}
+
+	private void renderIconsAndCounters(Graphics2D graphics, int x, int y)
+	{
+		final String counterText = Integer.toString(currentValue);
+		final int widthOfCounter = graphics.getFontMetrics().stringWidth(counterText);
+		final int centerText = (WIDTH - PADDING) / 2 - (widthOfCounter / 2);
+
+		if (config.enableCounter())
+		{
+			graphics.setFont(FontManager.getRunescapeSmallFont());
+			textComponent.setColor(Color.WHITE);
```

Comment:
It looks like `TEXT_COMPONENT` is only ever set to be white; this should be moved into the constructor instead of being called every time this method is called while `config.enableCounter()` is true.